src/colmap/estimators/cost_functions.h

```diff
@@ -349,182 +349,192 @@ inline void EigenQuaternionToAngleAxis(const T* eigen_quaternion,
-                  const T* const j_from_world_q,
-                  const T* const j_from_world_t,
+  bool operator()(const T* const cam_from_world_rotation,
+                  const T* const cam_from_world_translation,
                   T* residuals_ptr) const {
-    const Eigen::Quaternion<T> j_from_i_q =
-        EigenQuaternionMap<T>(j_from_world_q) *
-        EigenQuaternionMap<T>(i_from_world_q).inverse();
-    const Eigen::Quaternion<T> param_from_measured_q =
-        j_from_i_q * i_from_j_.rotation.cast<T>();
-    EigenQuaternionToAngleAxis(param_from_measured_q.coeffs().data(),
-                               residuals_ptr);
-
-    Eigen::Matrix<T, 3, 1> i_from_jw_t =
-        i_from_j_.translation.cast<T>() - EigenVector3Map<T>(i_from_world_t);
-    Eigen::Map<Eigen::Matrix<T, 3, 1>> param_from_measured_t(residuals_ptr + 3);
-    param_from_measured_t =
-        EigenVector3Map<T>(j_from_world_t) + j_from_i_q * i_from_jw_t;
-
-    Eigen::Map<Eigen::Matrix<T, 6, 1>> residuals(residuals_ptr);
-    residuals.applyOnTheLeft(sqrt_information_j_.template cast<T>());
+    Eigen::Map<Eigen::Matrix<T, 3, 1>> residuals(residuals_ptr);
+    residuals = position_prior_in_world_.cast<T>() +
+                EigenQuaternionMap<T>(cam_from_world_rotation).inverse() *
+                    EigenVector3Map<T>(cam_from_world_translation);
+    residuals.applyOnTheLeft(
+        position_prior_in_world_sqrt_info_.template cast<T>());
     return true;
   }
 
  private:
-  const Rigid3d& i_from_j_;
-  const EigenMatrix6d sqrt_information_j_;
+  const Eigen::Vector3d position_prior_in_world_;
+  const Eigen::Matrix3d position_prior_in_world_sqrt_info_;
 };
 
-// Cost function for aligning one 3D point with a reference 3D point with
-// covariance. Convention is similar to colmap::Sim3d
-// residual = scale_b_from_a * R_b_from_a * point_in_a + t_b_from_a -
-// ref_point_in_b
-struct Point3dAlignmentCostFunctor {
+// 6-DoF error between two absolute camera poses based on a prior on their
+// relative pose, with identical scale for the translation. The covariance is
+// defined in the reference frame of the camera i. Its first and last three
+// components correspond to the rotation and translation errors, respectively.
+//
+// Derivation:
+//    j_T_w = ΔT_j·j_T_i·i_T_w
+//    where ΔT_j = exp(η_j) is the residual in SE(3) and η_j in tangent space.
+//    Thus η_j = log(j_T_w·i_T_w⁻¹·i_T_j)
+//    Rotation term: ΔR = log(j_R_w·i_R_w⁻¹·i_R_j)
+//    Translation term: Δt = j_t_w + j_R_w·i_R_w⁻¹·(i_t_j -i_t_w)
+struct RelativePosePriorCostFunctor {
  public:
-  Point3dAlignmentCostFunctor(const Eigen::Vector3d& ref_point,
-                              const Eigen::Matrix3d& covariance_point)
-      : ref_point_(ref_point),
-        sqrt_information_point_(SqrtInformation(covariance_point)) {}
+  RelativePosePriorCostFunctor(const Rigid3d& i_from_j_prior,
+                               const Eigen::Matrix6d& i_from_j_prior_cov)
```

Comment:
The name `i_from_j_prior_cov` does not make it clear in which reference frame the covariance is defined, how about `i_prior_cov` instead?